xpdan/tools.py

```diff
@@ -0,0 +1,169 @@
+import numpy as np
+from matplotlib.path import Path
+import scipy.stats as sts
+
+
+def margin(img_shape, edge_size):
+    """
+    Mask the edge of an image
+
+    Parameters
+    -----------
+    img_shape: tuple
+        The shape of the image
+    edge_size: int
+        Number of pixels to mask from the edge
+
+    Returns
+    --------
+    2darray:
+        The mask array, bad pixels are 0
+    """
+    mask = np.ones(img_shape, dtype=bool)
+    mask[edge_size:-edge_size, edge_size:-edge_size] = 0.
+    return ~mask
+
+
+def binned_outlier(img, r, alpha, bins, mask=None):
+    """
+    Generates a mask by identifying outlier pixels in bins and masks any
+    pixels which have a value greater or less than alpha * std away from the
+    mean
+
+    Parameters
+    ----------
+    img: 2darray
+        The  image
+    r: 2darray
+        The  array which maps pixels to bins
+    alpha: float or tuple or, 1darray
+        Then number of acceptable standard deviations, if tuple then we use
+        a linear distribution of alphas from alpha[0] to alpha[1], if array
+        then we just use that as the distribution of alphas
+    bins: list
+        The bin edges
+    mask: 1darray
+        A starting flattened mask
+
+    Returns
+    --------
+    2darray:
+        The mask
+    """
+
+    if mask is None:
+        working_mask = np.ones(img.shape).astype(bool)
+    else:
+        working_mask = mask.copy()
+    if working_mask.shape != img.shape:
```

Comment:
You could have a mask or an image which was raveled into a 1D array, so we at least try to make the shapes comparable. If that fails it will throw a numpy based exception.